client_v25_test.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-FileCopyrightText: © Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dgo_test
+
+import (
+	"testing"
+
+	"github.com/dgraph-io/dgo/v240"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestOpen(t *testing.T) {
+	var err error
+
+	_, err = dgo.Open("127.0.0.1:9180")
+	require.ErrorContains(t, err, "first path segment in URL cannot contain colon")
+
+	_, err = dgo.Open("localhost:9180")
+	require.ErrorContains(t, err, "invalid scheme: must start with dgraph://")
+
+	_, err = dgo.Open("dgraph://localhost:9180")
+	require.NoError(t, err)
+
+	_, err = dgo.Open("dgraph://localhost")
+	require.ErrorContains(t, err, "invalid connection string: host url must have both host and port")
+
+	_, err = dgo.Open("dgraph://localhost:")
+	require.NoError(t, err)
```

Comment:
Seems like this should error

Reply:
yeah, but it wasn't and I didn't want to explicitly check for this.